Common/Source/Dialogs/dlgMultiSelectList.cpp

```diff
@@ -279,175 +534,165 @@ static void OnMultiSelectListPaintListItem(WindowControl * Sender, LKSurface& Su
                 CAirspaceManager::Instance().GetSimpleAirspaceAltText(Comment1, sizeof (Comment1) / sizeof (Comment1[0]), airspace_copy.Base());
 
                 CAirspaceManager::Instance().AirspaceCalculateDistance((CAirspace*) pAS, &HorDist, &Bearing, &VertDist);
-                _stprintf(text2, TEXT("%3.1f%s (%s - %s)"), (double) HorDist*DISTANCEMODIFY, Units::GetDistanceName(), Comment1, Comment); //8 + 8+3   21
+                _sntprintf(text2, MAX_LEN, TEXT("%3.1f%s (%s - %s)"), (double) HorDist*DISTANCEMODIFY, Units::GetDistanceName(), Comment1, Comment); //8 + 8+3   21
 
                 /****************************************************************
                  * for drawing the airspace pictorial, we need the original data.
                  * copy contain only base class property, not geo data,
                  * original data are shared ressources !
                  * for that we need to grant all called methods are thread safe
                  ****************************************************************/
+                ShowTextEntries(Surface, rc,  text1, text2);
                 pAS->DrawPicto(Surface, rc);
+
             }
             break;
 
+#ifdef  FLARM_MS
+            /************************************************************************************************
+             * IM_FLARM
+             ************************************************************************************************/
+        case IM_FLARM:
+            LKASSERT(Elements[i].iIdx  < FLARM_MAX_TRAFFIC);
+            LKASSERT(Elements[i].iIdx  >= 0);
+            FLARM_TRAFFIC Target;
+            LockFlightData();
+              memcpy( &Target, &GPS_INFO.FLARM_Traffic[Elements[i].iIdx], sizeof(     FLARM_TRAFFIC));
+            UnlockFlightData();
+            pFlarm= ( FLARM_TRAFFIC* )  &Target;
+            LKASSERT(pFlarm);
+            BuildFLARMText(pFlarm,text1,text2);
+            ShowTextEntries(Surface, rc,  text1, text2);
+
+#ifdef FLARM_PICTO_THREADSAFE
+            MapWindow::DrawFlarmPicto(Surface, rc, pFlarm);   // draw MAP icons
+#else
+            DrawUTF8FlarmPicto(Surface, rc, pFlarm);          // use alternate UTF8 icons
+#endif
 
-        case IM_TASK_PT:
+            break;
+#endif // FLARM_MS
+
+#ifdef TEAM_CODE_MS
+            /************************************************************************************************
+             * IM_TEAM
+             ************************************************************************************************/
+        case IM_TEAM:
+          _sntprintf(text1,MAX_LEN,_T("%s:"),gettext(_T("_@M700_"))); //_@M700_ "Team code"
+          _sntprintf(text2,MAX_LEN,_T("%s"), CALCULATED_INFO.OwnTeamCode );
+            ShowTextEntries(Surface, rc,  text1, text2);
+            if(Appearance.UTF8Pictorials)
+              UTF8Pictorial( Surface,  rc, _T("⚑"), RGB_VDARKRED);
```

Comment:
string literal containing non usascii characters are not crossplatform safe, neet bo be moved in language file